picard/config.py

```diff
@@ -74,24 +73,16 @@ def __getitem__(self, name):
         return self.value(name, opt, opt.default)
 
     def __setitem__(self, name, value):
-        self.lock_for_write()
-        try:
-            key = self.key(name)
-            self.__qt_config.setValue(key, value)
-        finally:
-            self.unlock()
+        key = self.key(name)
+        self.__qt_config.setValue(key, value)
 
     def __contains__(self, name):
         return self.__qt_config.contains(self.key(name))
 
     def remove(self, name):
-        self.lock_for_write()
-        try:
-            if name in self:
-                key = self.key(name)
-                self.__qt_config.remove(key)
-        finally:
-            self.unlock()
+        if name in self:
+            key = self.key(name)
```

Comment:
Since `name in self` is using `__contains__` which is also calling `self.key(name)`, we could make it more efficient (and easier to understand imho):

```python
key = self.key(name)
config = self.__qt_config
if config.contains(key):
    config.remove(key)
```

Reply:
Good point, fixed that